pkg/rancher-desktop/main/extensions/extensions.ts

```diff
@@ -171,25 +178,84 @@ export class ExtensionImpl implements Extension {
 
     await Promise.all(paths.map(async(p) => {
       try {
-        await this.client.copyFile(this.id, p, binDir);
+        await this.client.copyFile(this.id, p, binDir, { namespace: this.extensionNamespace });
       } catch (ex: any) {
         throw new ExtensionErrorImpl(ExtensionErrorCode.FILE_NOT_FOUND, `Could not copy host binary ${ p }`, ex);
       }
     }));
   }
 
+  get containerName() {
+    const normalizedId = this.id.toLowerCase().replaceAll(/[^a-z0-9_-]/g, '_');
+
+    return `rd-extension-${ normalizedId }`;
+  }
+
+  /**
+   * Extra the Docker Compose files into a newly-created temporary directory.
+   * This is required because nerdctl doesn't seem to keep the definition around,
+   * @note The caller is expected to remove the output directory.
+   * @param composePath the value of metadata.vm.composefile
+   */
+  protected async extractComposeDefinition(composePath: string): Promise<string> {
+    const workDir = await fs.promises.mkdtemp(path.join(os.tmpdir(), 'rd-ext-install-'));
+    const composeDir = path.posix.dirname(path.posix.normalize(composePath));
+
+    await this.client.copyFile(
+      this.id,
+      composeDir === '.' ? '/' : `${ composeDir }/`,
+      workDir,
+      { namespace: this.extensionNamespace });
+
+    return workDir;
+  }
+
+  protected async installContainers(workDir: string, metadata: ExtensionMetadata): Promise<void> {
+    function isImage(input: any): input is { image: string } {
+      return typeof input?.image === 'string';
+    }
+
+    function isComposefile(input: any): input is { composefile: string } {
+      return typeof input?.composefile === 'string';
+    }
+
+    if (isImage(metadata.vm)) {
+      // eslint-disable-next-line no-template-curly-in-string -- literal ${DESKTOP_PLUGIN_IMAGE}
+      const imageID = metadata.vm.image === '${DESKTOP_PLUGIN_IMAGE}' ? this.id : metadata.vm.image;
+      const stdout = await this.client.run(imageID, {
+        namespace: this.extensionNamespace,
+        name:      this.containerName,
+        restart:   'always',
+      });
```

Comment:
I'm a little concerned about this, since the [docs on metadata.json](https://docs.docker.com/desktop/extensions-sdk/architecture/metadata/#vm-section) say:

> When you use image, a default compose file is generated for the extension.

That being said, at the moment I don't see any reason why this wouldn't work. It's just that the extensions docs are specific about the backend containers that run in the VM being `docker compose` services.

Reply:
Oh wow, I totally missed that bit. Given that we don't need this _right now_, I've filed #4268 instead of fixing it here.